SongBird/app/src/main/java/shuvalov/nikita/songbird/MainActivity.java

```diff
@@ -0,0 +1,165 @@
+                            e.printStackTrace();
+                        }
+
+                        @Override
+                        public void onResponse(Call call, Response response) throws IOException {
+                            if (!response.isSuccessful()) {
+                                throw new IOException();
+                            }
+//                            Gson tweetGson = new Gson();
+                            try {
+                                JSONArray listOfTweets = new JSONArray(response.body().string());
+                                mTweetInfo.clear();
+                                for (int i =0; i<listOfTweets.length();i++){
+                                    mTweetInfo.add(new Gson().fromJson(String.valueOf(listOfTweets.getJSONObject(i)),TweetInfo.class));
+                                }
+                            } catch (JSONException e) {
+                                e.printStackTrace();
+                            }
+                            MainActivity.this.runOnUiThread(new Runnable() {
+                                @Override
+                                public void run() {
+                                    mAdapter.notifyDataSetChanged();
+                                }
+                            });
+                        }
+                    });
+
+                }else{
+                    Toast.makeText(MainActivity.this, "Hold your horses", Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+
+    }
+
+    private class BearerTokenTask extends AsyncTask<String, Void, BearerTokenResponse>{
+
+        @Override
+        protected BearerTokenResponse doInBackground(String... strings) {
+            Log.d("MALLET", "Working");
+            OkHttpClient okHttpClient = new OkHttpClient();
+            Headers header = new Headers.Builder()
+                    .add("Authorization"," Basic " +strings[0])
+                    .add("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8")
+                    .build();
+            FormBody body = new FormBody.Builder()
+                    .add("grant_type", "client_credentials")
+                    .build();
+            Request request = new Request.Builder()
+                    .post(body)
+                    .headers(header)
+                    .url("https://api.twitter.com/oauth2/token")
+                    .build();
+            try {
+                Gson gson = new Gson();
+                Response response = okHttpClient.newCall(request).execute();
+                return gson.fromJson(response.body().string(),BearerTokenResponse.class);
+
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+            return null;
+        }
+
+        @Override
+        protected void onPostExecute(BearerTokenResponse s) {
+            super.onPostExecute(s);
+            mBearerTokenHolder = s;
```

Comment:
Since this code doesn't affect the views at all, technically you could just put it at the end of your `doInBackground()` method and not bother overriding `onPostExecute()`